src/Orckestra.Composer/Providers/Localization/LocalizationExtensions.cs

```diff
@@ -36,7 +38,7 @@ public static string GetLocalizedErrorMessage(this ILocalizationProvider localiz
         /// <param name="price">The price.</param>
         /// <param name="cultureInfo">The culture information.</param>
         /// <returns></returns>
-        public static string FormatPrice(this ILocalizationProvider localizationProvider, decimal price, CultureInfo cultureInfo)
+        public static string FormatPrice(this ILocalizationProvider localizationProvider, decimal price, CultureInfo cultureInfo, string currencyCode = default)
```

Comment:
To avoid duplicate of code can you please remove here parameter string currencyCode = default
You added separate method FormatPrice(this ILocalizationProvider localizationProvider, decimal price, string currencyCode)
So call this method instead in places where you need 
Inside this new method calculate culture based on Currency and call existent method
FormatPrice(...decimal price, CultureInfo cultureInfo)